docs/source/overview/index.rst

```diff
@@ -0,0 +1,53 @@
+Overview (for Programmers)
+===================================
+
+This page provides a broad overview of how the model works. It is intended for programmers who want to read and/or edit the code.
+
+.. highlight:: bash
+
+
+Source Code and Packages
+---------------------------------
+
+The main package for the source code is ``microsim`` and the entry-point for the program is the ``main.py`` file (i.e. ``microsim/main.py``). To run the model and see what arguments are available, execute the following from the root project directory:
+
+    $ python microsim/main.py --help
+
+(for full details about running the model see the :doc:`usage guide<../usage/index>`).
+
+Regardless of whether the Python or OpenCL versions of the model are executed, ``main.py`` wil do the following things:
+
+1. Process the command line arguments
+
+2. Load the data. These can either be read from a cache that was created previously or else the raw files need to be read. The :ref:`population_initialisation.py<population_initialisation-label>` file reads the data. It creates a pandas DataFrame called ``individuals``, which holds the characteristics of every individual in the study area, and a dictionary called ``activity_locations`` which points to DataFrames for the locations that people will visit (e.g. home, school, retail, etc.)
```

Comment:
It would be good if we could expand on what the raw source data files are. It may be more appropriate to create a data sources section within this Developer guide part of the docs to do that.